engine/src/damage/damageable_object.cpp

```diff
@@ -28,18 +30,27 @@ DamageableObject::DamageableObject(std::vector<DamageableLayer> layers,
 }*/
 
 
-void DamageableObject::DealDamage( const CoreVector &attack_vector, Damage &damage ) {
-    for(DamageableLayer layer : layers) {
-        layer.DealDamage(attack_vector, damage);
+InflictedDamage DamageableObject::DealDamage( const CoreVector &attack_vector, Damage &damage ) {
+    InflictedDamage inflicted_damage(3); // Currently hard-coded default is 3!
 
-        if(layer.core_layer) {
+    // Higher index layers are outer layers. We therefore need to reverse the order.
+    for (DamageableLayer& layer : boost::adaptors::reverse(layers)) {
+        std::cout << "Damaging layer " << layer.layer_index << "\n";
```

Comment:
Can this be changed to use the Boost logging please?

Reply:
yes, we should have any calls to `std::cout`  or `std::cerr`.